libknet/internals.h

```diff
@@ -409,9 +409,14 @@ typedef struct knet_transport_ops {
  * and fd_tracker read lock (from RX thread)
  */
 	int (*transport_rx_is_data)(knet_handle_t knet_h, int sockfd, struct knet_mmsghdr *msg);
-} knet_transport_ops_t;
 
-socklen_t sockaddr_len(const struct sockaddr_storage *ss);
+/*
+ * this function is called by links.c when a link down event is recorded
+ * to notify the transport that packets are not going through, and give
+ * transport the possibility to take actions.
```

Comment:
Minor English quibble here. "possibility" should be "opportunity"

Reply:
Otherwise it all looks fine and works for me. I did have one socket that looked like it had a large backlog after taking an IF down but it cleared after a minute or so.

Reply:
fixed